### DISCUSSION TITLE
Does CanvasBlocker needed when using Firefox 120 with ETP-Strict?

### DISCUSSION THREAD

**Discussion by sha-265** [kkapsner/CanvasBlocker#676] [Q&A]
According to [Firefox 120 changelog](https://www.mozilla.org/en-US/firefox/120.0/releasenotes/), Mozilla added Canvas API fingerprint protection when using ETP in strict mode:
> Firefox’s private windows and ETP-Strict privacy configuration now enhance the Canvas APIs with Fingerprinting Protection, thereby continuing to protect our users’ online privacy.

Do we need CanvasBlocker when using ETP-Strict? @kkapsner

**Comment by User3409** [comment]
Last time i needed both canvasblocker + resistfingerprinting option enabled to confuse fingerprint.com

**Reply by sha-265** [reply]
It's not what I asked. I don't use resistfingerprinting anyway.

**Reply by User3409** [reply]
I do not think ETP prevents fingerprinting. Isn't ETP rather based on blocklists?

**Reply by sha-265** [reply]
Read my main post again. ETP has different features, not all of them using blocklists.

**Comment by kkapsner** [comment]
CB provides more functionality and therefore is still a valid tool. But ETP is as well. They can work hand in hand.

**Reply by sha-265** [reply]
But what about this specific feature? Do we need it if ETP strict is enabled?

**Reply by kkapsner** [reply]
It depends on your use case. As said: CB provides more functionality and more control how the CB protection exactly works.

**Comment by Wakagithub** [comment]
I'm just a User, but based from recent experience, CB is quite effective to at least temporarily give a User enough time. A few months ago I was falsely flagged by Reddit, maybe "spam" which I think is their default (with only their AI replying the same on my appeal), so I lost the ability to participate there. But I really need answers and experiences from Users (actual people) from around the world.

So I was able to create a temporary account, ask my question, and then get an answer. Before CB, workarounds like VPN's, even FF's own ETP, using different browsers, and similar ideas by themselves or combos of them didn't work, like fast, automatic suspension just within 24 hours, which made me realize spying is indeed true and powerful, and that's there's no privacy.

Still, I didn't want to take any chances, so I deleted that temp account within 1 week (remember that within 24 hrs, my temp Reddit account would be suspended immediately without CB back then), and hence I would just create another temporary one when the time comes I need to again. Of course I would not use it for mere commenting, but for utmost need only.

**Comment by sha-265** [comment]
I wonder if the answer for this [question](https://www.reddit.com/r/firefox/comments/1ozc955/is_canvasblocker_obsolete_now/) changed, after FF recent canvas fingerprinting protection.

If I'm just using CB in it's default configuration without any special features usage, do I need it with ETP strict enabled?

**Reply by kkapsner** [reply]
It still provides additional value as it has a different approach.

**Reply by sha-265** [reply]
Different approach to solve the same problem? I don't need two solutions for the same problem, only one

**Reply by kkapsner** [reply]
The problem is the same: tracking. I have not looked into the details of ETP lately but I think there can be cases where it would not protect but CB would.
